### PR TITLE
Feature/kan 25 backend jwt authentication

### PR DESCRIPTION
**Pull request by RezaBG**

### CODE REVIEW

**Comment by IshanRastogi** [comment]
The `branch name` doesn't match with the `work done` in the branch. Are you sure that you've completed the tasks that you wanted to ?? 

According to the branch name: you should be doing some `backend jwt authentication` here, however, from the looks of it the only thing accomplished here is the fixing of your `TaskStatus` and associated `case` related issues.

**Comment by RezaBG** [comment]
- TaskStatus and Case Handling: Addressed and fixed issues related to TaskStatus and case normalization for task statuses.
- Endpoint Testing: Successfully tested all 14 endpoints, ensuring they behave as expected.

File: tests/test_insert_task.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from sqlalchemy.orm import Session
+from src.models import Task, TaskStatus
+
+def test_insert_task(db: Session):
```
Comment (IshanRastogi) [inline tests/test_insert_task.py:4]: can you run this individual task and check the result ?

**Review by IshanRastogi — APPROVED** [review]
Merge this branch and then let's debug the next bit.

**Pull request merged**